--- xalan-c-1.11/c/src/xalanc/Makefile.in	2012-03-19 18:18:40.000000000 +0200
+++ xalan-c-1.11/c/src/xalanc/Makefile.in.orig	2017-01-13 14:06:16.000000000 +0200
@@ -123,7 +123,7 @@
     Harness \
 	XPathCAPI
 
-ALL_BINARIES = Xalan$(EXT)
+ALL_BINARIES =
 
 include $(XALANCROOT)/version.incl
 include ../../Makefile.incl
@@ -151,7 +151,7 @@
 ${XSL_OBJ_DIR}/%.o:%.hpp
 	$(CC1) $(XSL_BUILD_OPTIONS) -c $(XSL_INCL) $(EXTRA_COMPILE_OPTIONS) -o $@ $(patsubst .hpp,.cpp,$<)
 
-all:	 lib  testXSLT testXPath XalanExe
+all:	 lib
 
 prepare:
 	@echo Preparing the directory structure for a build ...
@@ -243,7 +243,7 @@
 endif ## OS390                                
 endif ## OS390BATCH
 
-install: lib XalanExe
+install: lib
 	$(MAKE) -C Utils $@
 	$(MKINSTALLDIRS) $(DESTDIR)/$(bindir)
 	for bin in $(ALL_BINARIES); do \
--- xalan-c-1.11.orig/c/Makefile.incl.in	2012-10-14 23:54:01.000000000 +0300
+++ xalan-c-1.11/c/Makefile.incl.in	2017-11-23 15:40:17.000000000 +0200
@@ -87,7 +87,7 @@
 # Library names
 LIB_NAME = lib$(PROJECT_NAME)
 LIB = $(LIB_NAME)$(SHLIBSUFFIX).$(LIB_MAJOR_VER).$(LIB_MINOR_VER)
-SO_NAME = $(LIB_NAME)$(SHLIBSUFFIX).$(LIB_MAJOR_VER)
+SO_NAME = $(LIB_NAME)$(SHLIBSUFFIX)
 LINK_NAME = $(LIB_NAME)$(SHLIBSUFFIX)
 XALAN_LIB_PATH = -L$(XSL_LIB_DIR)
 LIBRARY_LINK_NAME = -l$(PROJECT_NAME)
@@ -97,7 +97,7 @@
 LOC_PROJECT_NAME = xalanMsg
 LOC_LIBNAME = lib${LOC_PROJECT_NAME}
 LOC_LIB = ${LOC_LIBNAME}$(SHLIBSUFFIX).${LIB_MAJOR_VER}.${LIB_MINOR_VER}
-LOC_SONAME = ${LOC_LIBNAME}${SHLIBSUFFIX}.${LIB_MAJOR_VER}
+LOC_SONAME = ${LOC_LIBNAME}${SHLIBSUFFIX}
 MESSAGELIB = ${XSL_LIB_DIR}/${LOC_LIB}
 EXTRA_LOC_LINK_OPTIONS = $(EXTRA_LINK_OPTIONS)
 
