--- xml-security-c-1.7.2/configure	2016-06-19 14:05:34.000000000 +0300
+++ xml-security-c-1.7.2/configure.bkp	2016-06-19 14:08:33.000000000 +0300
@@ -15342,8 +15342,8 @@
   version_type=linux # correct to gnu/linux during the next big refactor
   need_lib_prefix=no
   need_version=no
-  library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
-  soname_spec='${libname}${release}${shared_ext}$major'
+  library_names_spec='$libname${shared_ext}'
+  soname_spec='${libname}${shared_ext}'
   finish_cmds='PATH="\$PATH:/sbin" ldconfig -n $libdir'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
